### ISSUE TITLE
W0120: "return" is not detected

### ISSUE THREAD

**Issue by regisz** [PyCQA/pylint#2490]
### Situation
```
def my_func():
  for my_var in [1, 2, 3]:
    if my_var == 2:
      return True
  else:
    return False
```

### Current behavior
`W0120` occurs because there is no `break` in loop.

### Expected behavior
This warning is fake in this case.

### pylint --version output
I use pymode, and it uses this pylint hash: 66cb321.
I do not use pylint directly.

**Comment by PCManticore** [comment]
This has occurred in the past a couple of times. As mentioned in my previous comments on this matter, feel free to remove the `else` clause from the `for` loop as it currently superfluous since the semantics of `for else` only makes sense on having a `break` inside the loop itself.
